Makefile.generated_files

```diff
@@ -287,6 +287,118 @@ $(DEEPCOPY_GEN):
 	touch $@
 
 #
+# Defaulter generation
+#
+# Any package that wants defaulter functions generated must include a
+# comment-tag in column 0 of one file of the form:
+#     // +k8s:defaulter-gen=<VALUE>
+#
+# The <VALUE> depends on context:
+#     on types:
+#       true:  always generate a defaulter for this type
+#       false: never generate a defaulter for this type
+#     on functions:
+#       covers: if the function name matches SetDefault_NAME, instructs
```

Comment:
Not random - used elsewhere in the code to describe the distinction between:

1. A defaulting function that only defaults a particular struct but not its children (non-covering)
2. A defaulting function that "covers" defaulting for itself and all nested children.

Our existing defaulting functions are non-covering - you have to call them and all nested functions to be safe.  The new generated ObjectDefaulters are covering - you don't have to call nested functions.  Some of our top level ObjectDefaulters call functions that are covering as well (other ObjectDefaulters).  I wanted to enable the specific use case of someone writing a custom defaulter that completely handles the types underneath it (overriding the generic defaulters defined in other packages).